src/runtime/Component.tsx

```diff
@@ -130,20 +142,31 @@ export abstract class Component<
     this.clearPrevState();
 
     try {
-      const entity = this.createEntity();
+      const asyncFunction = this.createEntity.bind(this);
+
+      // if (!this._entity) {
```

Comment:
I just realized this error too... it's because the entity isn't being recreated when the props change. So for example if you change the address of a DAO, the entity isn't being recreated. Since "createEntityWithProps" will only ever be called when the props change, we should always be recreating the entity. So this if condition so go away. Nice catch! Sorry about that :P